### DISCUSSION TITLE
Possible Power Supply Failure? Does not fully boot on mains power, but does on USB.

### DISCUSSION THREAD

**Discussion by NathanVanCleave** [HASwitchPlate/HASPone#225] [Troubleshooting]
I have a pre-assembled HASP I purchased from the Tindie store in September of 2021. It has been working flawlessly since then until about a week ago when my router (UDM non-pro) started having issues with some 2.4Ghz clients repeatedly disconnecting. The HASP was unable to get back on the wifi at one point overnight and so I power cycled it from the breaker the next morning. Ever since then, **it almost never fully boots**, usually getting stuck on "HASP Initiating" but sometimes making a connection to wifi (confirmed from the router) and even more rarely connected to MQTT (confirmed in Home Assistant). Even when it does manage to make connections it is unresponsive from both the web UI and MQTT. 

However, when powering the D1 Mini directly via the USB port it fully boots within just a few seconds every time without issue. 

Steps I have taken:
- Upgrading the ESP firmware via USB to 1.0.5, 1.0.6, and even back to 1.0.4 (I believe it was still on the firmware 1.x.x it came with beforehand because I haven't touched it since installation over a year ago). I cannot find firmware older than 1.0.3 to test.
- Upgrading the LCD firmware via SD card (after updating the ESP firmware first) using this link https://github.com/HASwitchPlate/HASPone/blob/main/Nextion_HMI/HASwitchPlate.tft
- Factory resetting the device
- Changing the LCD Serial Speed between 115200 and 921600
- Powering from mains from a different location
- Checking voltages: 120V from mains. D1 Mini 5.5V pin has 5.5V while on mains and 4.4V with USB.

That last step is the only issue I haven't figured out. Should I be seeing 4.4V instead of 5.5V on the D1 Mini while powered from mains? This is what's making me think it may be a power issue from the power supply. We've noticed a few lights in the house flicker a few times over the last couple of weeks (roughly since the wifi issues started). While we haven't had a power outage in months (only happens a few times a year), I'm wondering if there have been some power surges interrupting smaller 2.4Ghz wifi devices. I've been tracking disconnects from Tasmotized devices (mostly Sonoff S31's) and they go through periods of frequent disconnects (10-30 over a 4-6 hour period at various times during the day or night). Either way, I think I can rule out wifi issues on the HASP because it works perfectly when powered over USB.

Note that the device did fully boot on mains after one of the firmware upgrades (ESP to 1.0.5 and LCD) and stayed online for about 12 hours before disconnecting again and failing to reboot fully. Below is a screenshot of the most recent settings I'm using.
<img width="456" alt="Screenshot 2023-02-15 at 10 20 55 AM" src="https://user-images.githubusercontent.com/26234190/219139622-44fb16bd-6f11-4e88-8d00-34cc063b042e.png">

I have not tried capturing serial logs while connected to mains because I'm unsure about connecting to USB at the same time. If that's okay to do then let me know and I'll share what I find.

Thanks!

-Nathan

**Comment by NathanVanCleave** [comment]
Gah! I realized after typing that all up that my HASP, while powered via USB, went unresponsive. It's an issue I forgot to include and _may_ help to rule out power supply issues.

A few times since the start of this issue, I've noticed that the clock has stopped updating and realize that the HASP is unresponsive to touch and no longer connected to MQTT. It still shows as being connected to wifi from the router with no disconnects during that time, though the web UI is unresponsive. Power cycling from USB brings it back instantly. No other devices were having wifi or MQTT issues during this latest event.

**Comment by NathanVanCleave** [comment]
Okay, I might have raised this issue too soon. My router may actually be to blame. I spent a few days troubleshooting and reading everything on this github trying to solve it on my own, but didn't give myself quite enough time I guess. 

I had initially moved the HASP off my NoT VLAN (no internet access) and to my main SSID while shotgunning solutions. After many firmware flashes and many issues with just connecting to the hotspot I decided to try putting it back on the NoT VLAN it was originally on...and it magically started working on mains power again! UniFi seems to be kind of sticky about switching between subnets and actually forgetting devices, or more likely I just don't know enough about it. 

Note I had also flashed it to 1.0.1 (just before moving to the original VLAN) with the LCD Serial Speed at 115200 when it started working. I've since moved it to 1.0.6 and it's still running. I'm hoping that the wifi connection improvements in 1.0.6 might help mitigate future wifi issues that brought it down in the first place. 

I still don't understand why it worked on USB power and not mains, so I'm going to leave this discussion up for now in case it hasn't been fully resolved.

**Comment by NathanVanCleave** [comment]
Aaaaaand the issue remains. The HASP stayed connected overnight till this morning. Same behavior as before. No other devices were dropped from the network. Restarting the device via power cycle just gets stuck on HASP Initializing.

**Comment by aderusha** [comment]
Any chance you could set it up on USB and gather logs for a bit, see if you can catch the behavior in action?

**Reply by NathanVanCleave** [reply]
Can I power it from mains with the USB connected to my computer? I'm only seeing the issue on mains power.

**Reply by aderusha** [reply]
Sadly... not safely, no.  Possible if you have a way to isolate but I wouldn't recommend it unless you fully understood the implications.

Does running on USB power over a day or two work?

**Reply by NathanVanCleave** [reply]
Glad I asked. 

I have not tested USB power for an extended period of time, but I have observed the device go unresponsive within an hour after booting fine on USB. Which is similar to what happens when it has successfully booted on mains and stayed responsive for several hours. 

I will set it up on USB and observe it for a while. I'm using ESPHomeFlasher to see the serial logs. Is there a better program to capture logs over serial USB long-term than just leaving that open?

**Reply by aderusha** [reply]
PuTTY.exe is a common solution, I've been using [SimplySerial](https://github.com/fasteddy516/SimplySerial) lately as a great does-one-thing-well console application for monitoring serial coms.

**Reply by NathanVanCleave** [reply]
Thanks for the suggestions!

**Reply by NathanVanCleave** [reply]
I was able to use CoolTerm (Mac user) to capture logs over either USB or Telnet. Naturally, only the USB logs contain full boots and only on USB power. I think I need to continue testing, but figured I'd share what I have so far in case there's something useful. 

I'm still unable to reliably recreate/predict the two general issues I'm seeing. 
1. Stuck during boot sequence (often connected to the network but web UI is unreachable)
2. Device quietly goes offline (screen and buttons still working, Unifi still shows a connection, but MQTT and web UI are disconnected) 

However, I do always have issues with the AP after a factory reset. It takes multiple attempts to get to the wifi config page after which it works normally. The AP connection is just really unstable on all devices I've tried (Mac, iPhone, Ubuntu on old Dell laptop). 

Notes about the linked logs:
I had it working normally on USB power for about 6 hours until it started going offline. The first three sections of the logs show instances of it going offline, going offline and then a sucessfull restart, and thirdly a factory reset via web UI after which it succesfully boots but has issues connecting to the AP and then an unresponsive web UI. 
The fourth and final section show when the device went offline on mains power

Notably, I went 24 hours of the device working flawlessly on mains after having it go offline on USB a bunch (previously I saw the same behavior when powering from mains at my desk and when it was in the wall). I didn't change the firmware and a factory reset was the only difference from when it was going offline on USB. I power cycled it probably 30 times throughout the day and it never had an issue coming back online like it used to, until I did one more power cycle before bed and it went offline 10 minutes later. It's been working flawlessly with power cycles so far today. 

Link to logs with comments:
https://gist.github.com/NathanVanCleave/d757f7a35fc2e2b7635da42510d20f84

**Reply by NathanVanCleave** [reply]
Some more info. I'm starting to think the issue is with the power supply. I've seen our lights flicker lately and it seems likely the power supply may have been damaged. This might help explain the overall inconsistent behavior that's made it so difficult to replicate issues and troubleshoot, at least while on mains power which is when most failures happen. 

I've been able to make it go offline while on mains power by turning the brightness all the way up after it was all the way down. I noticed this after it went offline at the same time a few days, which is when the display is woken up and a lot of information on it is changed. I was able to replicate this in testing a few times and have since changed the automations to raise the brightness and update information on the display more slowly.

Even then it's been inconsistent with me being able to change the brightness while on mains power without it going offline. About the only behavior that has always been consistent is that when it goes offline it will never fully boot again on mains power until after I reflash the firmware over USB. Though, a bad power supply still doesn't explain the unstable AP mode and web UI often not working on USB power.

I have a friend that is going to help me replace the power supply since it's only 5 solder points and shouldn't be too difficult. Do you still recommend the Meanwell power supply listed in the wiki or is there another one I should try?

**Reply by aderusha** [reply]
To my knowledge, the only part that will work will be the MeanWell IRM-03-5 (note there's no "S" on the end which would be the surface mount model).  The footprint is unique to that part.  I'm surprised that it'd fail but it's not impossible, in general, these were selected as being a high-quality and safe power solution from a trusted brand.  You should be able to get one for ~ $10 shipped.

This might get you started!  https://www.findchips.com/search/irm-03-5

**Reply by NathanVanCleave** [reply]
Thanks for the info. I'll order the one you linked.

Do you have any other ideas based on the information I've provided? I'm not very knowledgeable with electronics, but I know power supply issues can often explain weird happenings in general and it seems like a replacement is the logical next step to troubleshoot given what I'm seeing. 

Here are a few more event logs:
https://gist.github.com/NathanVanCleave/5c4b66520383844da97d62dce9805be7

I'll continue to capture logs (as long as the Telnet/Web UI remains responsive) but I'm not sure what else I can try short of replacing the power supply or D1 Mini.

**Reply by aderusha** [reply]
It's hard to say from here, I certainly haven't seen this behavior between wall power and USB which is really odd.  Electrically, what happens is that the MaanWell unit takes AC and turns that into 5V DC which is fed into the 5V rail on the WeMos.  USB works the same way, it feeds the USB-supplied 5V to the same place.  On the WeMos itself, that is then converted to 3.3v to run the ESP.

So, if it's working from USB and not from the MeanWell...  that kind of only leaves the MeanWell as the rest of the power path is the same between USB and MW.

**Reply by NathanVanCleave** [reply]
Well, I appreciate your time given this issue seems to be specific to my device.

It has been in the wall on mains running smoothly since last night, which I think is the first time it's even made it that long in that setup and is likely because I removed any large screen brightness jumps. Even the Web UI and Telnet are responsive. I think I'll leave it there for as long as it stays online and if/when it fails again I'll replace the power supply and report the results back here after a few weeks.

Again, thank you for your attention and time. I know single-user issues don't have a great time ROI when other users' units aren't experiencing the same issue. This HASP is in my top 5 favorite devices in my smart home and provides an incredible amount of utility, most importantly keeping us informed of what's going on outside so we don't let our dogs out the front door at the wrong time. Meaning, I'm highly motivated to get it stable again like it was for 16 straight glorious months.

**Reply by NathanVanCleave** [reply]
Quick update: It's been three weeks now without a single issue! 

To reiterate for anyone stumbling upon this, the only changes I effectively made (apart from many firmware flashes) was removing an automation that cranks the brightness from 10-100% in the mornings. I now turn in up in steps over a minute or so. I also had a lot of display information changes via MQTT commands at the same time every day, but I doubt that mattered. I have a new power supply on hand in case something happens, but that's seems increasingly unlikely.

**Reply by NathanVanCleave** [reply]
It made it three months this time before it started to act up again.

I noticed it offline and tried power cycling it. The same behavior as before started where it gets stuck at various places during boot up; initializing, connecting to wifi or connecting to MQTT. Same issues on USB power so I went ahead and swapped the power supply. Same issues remained so I reflashed the D1 Mini and was able to get it to fully boot on mains power. Put it back in the wall and it lasted about a week before stalling again.

By stalling I mean it just freezes and the time on the display matches the last time it was seen by MQTT. This seems to still always happen in the morning when the display brightness is turned up. I'd added a few more steps in the brightness ramping, but maybe it's still just drawing too much too quickly. If I can get it back online I'll add more steps (right now it goes from 10% to 100% in 6 step increments 2 seconds apart) and will increase the time between steps.

@aderusha would it be worth trying a swap of the D1 Mini? That's the only thing I can think of to do now, but wanted to ask first because desoldering the power supply was quite the ordeal for me even with just five leads. 

I should also mention that the indent on that backside of the 3D-printed housing where the Shelly behind it in the wall was pushing against it has gotten worse. I'm still unsure if it's heat from the Shelly or the MeanWell, but I wedged the Shelly to not push on it now. There is still a few mm between the top of the MeanWell and the screen inside the housing so it's not pushing on anything yet. [Here's a link](https://imgur.com/a/0iG7v6K) to some comparison photos:
